cmd/rpcdaemon/commands/eth_call.go

```diff
@@ -312,59 +329,101 @@ func (api *APIImpl) GetProof(ctx context.Context, address libcommon.Address, sto
 	}
 	defer tx.Rollback()
 
-	blockNr, _, _, err := rpchelper.GetBlockNumber(blockNrOrHash, tx, api.filters)
+	blockNr, blockHash, _, err := rpchelper.GetBlockNumber(blockNrOrHash, tx, api.filters)
 	if err != nil {
 		return nil, err
 	}
 
+	header := rawdb.ReadHeader(tx, blockHash, blockNr)
+	if header == nil {
+		return nil, err
+	}
+
 	latestBlock, err := rpchelper.GetLatestBlockNumber(tx)
 	if err != nil {
 		return nil, err
-	} else if blockNr != latestBlock {
-		return nil, fmt.Errorf(NotImplemented, "eth_getProof for block != latest")
-	} else if len(storageKeys) != 0 {
-		return nil, fmt.Errorf(NotImplemented, "eth_getProof with storageKeys")
-	} else {
-		addrHash, err := common.HashData(address[:])
-		if err != nil {
-			return nil, err
-		}
+	}
+
+	if latestBlock < blockNr {
+		// shouldn't happen, but check anyway
+		return nil, fmt.Errorf("block number is in the future")
+	}
 
-		rl := trie.NewRetainList(0)
-		rl.AddKey(addrHash[:])
+	if len(storageKeys) != 0 {
+		return nil, fmt.Errorf(NotImplemented, "eth_getProof with storageKeys")
+	}
 
-		loader := trie.NewFlatDBTrieLoader("getProof")
-		if err := loader.Reset(rl, nil, nil, false); err != nil {
-			return nil, err
-		}
+	addrHash, err := common.HashData(address[:])
+	if err != nil {
+		return nil, err
+	}
 
-		var accProof accounts.AccProofResult
-		accProof.Address = address
+	rl := trie.NewRetainList(0)
+	rl.AddKey(addrHash[:])
 
-		// Fill in the Account fields here to reduce the code changes
-		// needed in turbo/trie/hashbuilder.go
-		reader, err := rpchelper.CreateStateReader(ctx, tx, blockNrOrHash, 0, api.filters, api.stateCache, api.historyV3(tx), "")
-		if err != nil {
-			return nil, err
+	var loader *trie.FlatDBTrieLoader
+	if blockNr < latestBlock {
+		if latestBlock-blockNr > maxGetProofRewindBlockCount {
+			return nil, fmt.Errorf("requested block is too old, block must be within %d blocks of the head block number (currently %d)", maxGetProofRewindBlockCount, latestBlock)
 		}
-		a, err := reader.ReadAccountData(address)
+		tmpDir, err := os.MkdirTemp("", "eth_getHash")
```

Comment:
i advise youse existing `tmpdir`: no reason to manually create/remove folders/files - memdb/etl - will automatically delete what need - if `defer rollback/close`.

Reply:
I didn't like this part of the PR, but, wasn't really sure of the best way to do it and didn't see any of the other RPC daemon commands referencing a tempdir.

My best guess is that you're referring to the datadir temp dir?  I see there is a datadir on the CLI flags, so, I've pushed a commit on top that wires the `datadir.Dirs` through to the call.  Unfortunately this creates a bit of churn in the other tests, but, it's not too bad.

I'm not sure if that's what you were hoping for, so please let me know if you had something different in mind, happy to take a different approach.